griptape/artifacts/boolean_artifact.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import annotations
+from typing import Any
+from attrs import define, field
+from griptape.artifacts import BaseArtifact
+
+
+@define
+class BooleanArtifact(BaseArtifact):
+    value: bool = field(converter=bool, metadata={"serializable": True})
+    meta: dict[str, Any] = field(factory=dict, kw_only=True, metadata={"serializable": True})
+
+    @classmethod
+    def parse_bool(cls, value: str) -> BooleanArtifact:
+        """
+        Convert a string literal to a BooleanArtifact. The string must be either "true" or "false" with any casing.
+        """
+        if value is not None:
+            if value.lower() == "true":
+                return BooleanArtifact(True)
+            elif value.lower() == "false":
+                return BooleanArtifact(False)
+        raise ValueError(f"Cannot convert string literal '{value}' to BooleanArtifact")
```

Comment:
should this return `None` or raise an exception?

Reply:
I vote for raise an exception.

If the user's code assumes that the parse always succeeds and then it fails, returning None will (most likely) cause an exception to be raised by code that consumes this output. Then it will be the user's responsibility to trace it back to here, so we might as well do it for them.